### PR TITLE
Update nodeSelector and tolerations

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This allows the `cloud-controller-manager` to start on nodes with `etcd` taints by default. Additionally, allows configuration for additional taints if the user chooses. Currently this customization is not available at all.

Additionally allows configuration for `nodeSelector`. This is similar to how the `cloud-provider-aws` configures their Helm chart. https://github.com/kubernetes/cloud-provider-aws/blob/5db84490841ebd1eede368d03eb03c27dc556233/charts/aws-cloud-controller-manager/values.yaml#L16

#### Which issue(s) this PR fixes:

Fixes #
replace https://github.com/kubernetes-sigs/cloud-provider-azure/pull/3296

#### Special notes for your reviewer:
Thank you!

#### Does this PR introduce a user-facing change?

```release-note
- Add tolerations in values for cloud-controller-manager in Helm chart
- Add nodeSelector in values for cloud-controller-manager in Helm chart
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: helm/repo/index.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@ apiVersion: v1
 entries:
```
Comment (jwtty) [inline helm/repo/index.yaml:2]: Should we modify the existing index or create a new one?

Comment (lzhecheng) [reply helm/repo/index.yaml:2]: So far, we always use this one.

**Review by feiskyer — APPROVED** [review]
looks good

**Pull request merged**